zipline/pipeline/loaders/equity_pricing_loader.py

```diff
@@ -81,12 +81,16 @@ def load_adjusted_array(self, columns, dates, assets, mask):
             dates,
             assets,
         )
-        adjusted_arrays = [
-            AdjustedArray(raw_array, mask, col_adjustments)
-            for raw_array, col_adjustments in zip(raw_arrays, adjustments)
-        ]
 
-        return dict(zip(columns, adjusted_arrays))
+        out = {}
+        for c, c_raw, c_adjs in zip(columns, raw_arrays, adjustments):
+            out[c] = AdjustedArray(
+                c_raw.astype(c.dtype),
+                mask,
+                c_adjs,
+                c.missing_value,
+            )
+        return out
```

Comment:
I wrote this as a comprehension and then rewrote it this way because I thought it was easier to follow.  Going to leave it as is.